diff --git a/CMakeLists.txt b/CMakeLists.txt
index 365c023..a5bb204 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1561,3 +1561,5 @@ if(${CMAKE_GENERATOR} MATCHES "^Xcode.*|^Visual Studio.*")
 	file(GLOB_RECURSE SRC_FILES *.h *.cpp *.c *.cc *.mm)
 	source_group(TREE ${CMAKE_CURRENT_SOURCE_DIR} FILES ${SRC_FILES})
 endif()
+
+set_target_properties(flycast PROPERTIES DISABLE_PRECOMPILE_HEADERS ON)
